File: src/main/java/com/team3/gudit/auth/oauth2/OAuth2UserInfoFactory.java
package com.team3.gudit.auth.oauth2;

import java.util.Map;

public class OAuth2UserInfoFactory {
    private OAuth2UserInfoFactory() {}

    public static OAuth2UserInfo of (AuthProvider provider, Map<String, Object> attributes) {
        return switch (provider) {
            case KAKAO -> new KakaoUserInfo(attributes);
        };
    }
}

File: src/main/kotlin/com/team3/gudit/auth/oauth2/OAuth2UserInfoFactory.kt
package com.team3.gudit.auth.oauth2

object OAuth2UserInfoFactory {
    @JvmStatic
    fun of(provider: AuthProvider, attributes: Map<String, Any>): OAuth2UserInfo = when (provider) {
        AuthProvider.KAKAO -> KakaoUserInfo(attributes)
    }
}
